### PR TITLE
Add doctest 1.1.4

### PR DESCRIPTION
**Pull request by piribes**

### CODE REVIEW

File: cmake/projects/doctest/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+include(hunter_cacheable)
+
+# List of versions here...
+hunter_add_version(
+    PACKAGE_NAME
+    doctest
+    VERSION
+    "1.1.4"
+    URL
+    "https://github.com/piribes/doctest/archive/hunter.tar.gz"
```
Comment (ruslo) [inline cmake/projects/doctest/hunter.cmake]: Please use stable link here, I guess it should be `https://github.com/piribes/doctest/archive/1.1.4.tar.gz`.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1480
* https://travis-ci.org/ingenue/hunter/builds/216173481

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1482
* https://travis-ci.org/ingenue/hunter/builds/216176997

**Comment by ruslo** [comment]
Released:
* https://github.com/ruslo/hunter/releases/tag/v0.18.29

Feel free to add `pkg.doctest` wiki page and to update list:
* https://docs.hunter.sh/en/latest/packages/all.html

**Comment by piribes** [comment]
Done.

- https://github.com/ruslo/hunter/wiki/pkg.doctest
- #730

**Comment by piribes** [comment]
This use an hunterized version of doctest.
PR [doctest#60](https://github.com/onqtam/doctest/pull/60) waiting to be merged.

**Pull request merged**